app/src/main/java/protect/card_locker/AboutActivity.kt

```diff
@@ -0,0 +1,102 @@
+package protect.card_locker
+
+import android.os.Bundle
+import android.view.MenuItem
+import android.view.View
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import protect.card_locker.databinding.AboutActivityBinding
+
+class AboutActivity : CatimaAppCompatActivity() {
+    private var binding: AboutActivityBinding? = null
+    private var content: AboutContent? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val binding = AboutActivityBinding.inflate(layoutInflater)
+            .also { this.binding = it }
+        val content = AboutContent(this)
+            .also { this.content = it }
```

Comment:
Could you explain what this does or link me to a good resource explaining it? I'm pretty new to Kotlin and the explanations I'm finding online don't really explain this well to me. Given I will have to maintain this code, I need to understand what it does.

Reply:
Okay so:

Call `AboutContent`'s constructor and assign whatever it gives back to the variable `content`. Then set `this.content` to the newly created `content` variable. Right?

I don't quite understand why that's cleaner than just directly doing:
```
this.content = AboutContent(this)
```
though. Is this some kind of best practice I don't know about?

Reply:
It is to use a not nullable variable. in the this scope it is nullable. You would unnecessarily check if it null, after instantiating